test/integration/Client/client_lib_hybrid_argobots_test.cpp

```diff
@@ -10,14 +10,50 @@
 
 chronolog::Client*client;
 
+// Define bitmask for thread states
+enum ThreadState
+{
+    THREAD_INITIALIZED = 1 << 0, // 0001
+    CONNECTED = 1 << 1, // 0010
+    DISCONNECTED = 1 << 2  // 0100
+};
+
+std::vector <int> shared_state; // Global array of thread states
+std::mutex state_mutex;
+std::condition_variable state_cv;
+
 struct thread_arg
 {
     int tid;
 };
 
+// Function to update shared state and wait for all threads to reach a specific state
+void update_shared_state_and_wait(int tid, int target_state_bitmask)
+{
```

Comment:
I suggest that each thread updates only its own state. This way we know that the only writer on the shared_state[i] is thread i , every other thread has read only access to shared_state[i] making the mutex unnecessary.